core/src/main/java/io/questdb/cutlass/line/array/AbstractArray.java

```diff
@@ -94,6 +134,61 @@ public void close() {
         closed = true;
     }
 
+    /**
+     * Reshapes the array to the specified dimensions.
+     * <p>This method allows changing the array's shape after creation, enabling
+     * dynamic resizing based on runtime requirements.
+     * <br>
+     * Note: It resets the append position to the start of the array.
+     *
+     * @param shape the new dimensions for the array
+     * @throws IllegalStateException    if the array is already closed
+     * @throws IllegalArgumentException if shape has invalid dimensions
+     */
+    public void reshape(int... shape) {
```

Comment:
we are overusing ... syntax here, this makes sender allocating. Can we please review ... uses and create overloads for up to 3d arrays. In practice, order books are 1-d to 2-d structures, allocating arrays here is unneccessary

Reply:
I've talked about this with Jara. You can pre-allocate an `int[]` and call `reshape` with it, Java will accept that as well. `int...` is just extra convenience on top of this, allowing you to just put the numbers in.

Reply:
why don't we overload for up to 3 dims? and all over 3 can go into the ellipsis